src/s_stuff.h

```diff
@@ -185,6 +193,16 @@ EXTERN void sys_closesocket(int fd);
 typedef void (*t_fdpollfn)(void *ptr, int fd);
 EXTERN void sys_addpollfn(int fd, t_fdpollfn fn, void *ptr);
 EXTERN void sys_rmpollfn(int fd);
+#ifdef THREADED_IO
+#define t_rbskt struct _rbskt
+typedef void (*t_fdsendrmfn)(void *ptr);
+EXTERN void sys_addpollrb(int fd, int preserve_boundaries);
+EXTERN t_rbskt* sys_getpollrb(int fd);
+void sys_addsendfdrmfn(int sockfd, t_fdsendrmfn, void* x);
+ssize_t sys_sendto(int sockfd, const void *buf, size_t len, int flags, void* addr, size_t addrlen);
+EXTERN int rbskt_recv(t_rbskt* rbskt, char* buf, size_t length, void* nothing);
```

Comment:
I guess this will become `int rbskt_recv(t_rbskt* rbskt, char* buf, size_t length, void* sockaddr, socklen_t *addrlen);` to match the semantics of `recvfrom`? Shouldn't be too hard to transmit the socket address. After all, you already do it for sending.

Reply:
yes there is a TODO about that in `x_net.c`

Reply:
8591b3c10439eea17aad59fb0cc7bc61ff0681fb